layers/core_checks/cc_pipeline_graphics.cpp

```diff
@@ -1991,6 +1906,77 @@ bool CoreChecks::ValidateGraphicsPipelineRasterizationState(const vvl::Pipeline
+                if (!pipeline.IsDepthStencilStateDynamic() || !IsExtEnabled(device_extensions.vk_ext_extended_dynamic_state3)) {
+                    skip |= LogError("VUID-VkGraphicsPipelineCreateInfo-renderPass-09028", rp_state.Handle(), ds_loc,
+                                     "is NULL when rasterization is enabled "
+                                     "and subpass %" PRIu32 " uses a depth/stencil attachment.",
+                                     pipeline.Subpass());
+                }
+            } else if (ds_state->depthBoundsTestEnable == VK_TRUE) {
+                if (!enabled_features.depthBounds) {
+                    skip |= LogError("VUID-VkPipelineDepthStencilStateCreateInfo-depthBoundsTestEnable-00598", device,
+                                     ds_loc.dot(Field::depthBoundsTestEnable),
+                                     "depthBoundsTestEnable is VK_TRUE, but depthBounds feature was not enabled.");
+                }
+
+                if (!IsExtEnabled(device_extensions.vk_ext_depth_range_unrestricted) &&
+                    !pipeline.IsDynamic(CB_DYNAMIC_STATE_DEPTH_BOUNDS)) {
+                    const float minDepthBounds = ds_state->minDepthBounds;
+                    const float maxDepthBounds = ds_state->maxDepthBounds;
+                    if (!(minDepthBounds >= 0.0) || !(minDepthBounds <= 1.0)) {
+                        skip |= LogError("VUID-VkGraphicsPipelineCreateInfo-pDynamicStates-02510", device,
+                                         ds_loc.dot(Field::minDepthBounds),
+                                         "is %f, depthBoundsTestEnable is VK_TRUE, but VK_EXT_depth_range_unrestricted extension "
+                                         "is not enabled (and not using VK_DYNAMIC_STATE_DEPTH_BOUNDS).",
+                                         minDepthBounds);
+                    }
+                    if (!(maxDepthBounds >= 0.0) || !(maxDepthBounds <= 1.0)) {
+                        skip |= LogError("VUID-VkGraphicsPipelineCreateInfo-pDynamicStates-02510", device,
+                                         ds_loc.dot(Field::minDepthBounds),
+                                         "is %f, depthBoundsTestEnable is VK_TRUE, but VK_EXT_depth_range_unrestricted extension "
+                                         "is not enabled (and not using VK_DYNAMIC_STATE_DEPTH_BOUNDS).",
+                                         maxDepthBounds);
+                    }
+                }
+            }
+        }
+    }
+
+    // If subpass uses color attachments, pColorBlendState must be valid pointer
+    if (pipeline.fragment_output_state && !pipeline.fragment_output_state->color_blend_state &&
+        !pipeline.IsColorBlendStateDynamic()) {
+        uint32_t color_attachment_count = 0;
+        for (uint32_t i = 0; i < subpass_desc.colorAttachmentCount; ++i) {
+            if (subpass_desc.pColorAttachments[i].attachment != VK_ATTACHMENT_UNUSED) {
+                ++color_attachment_count;
```

Comment:
```suggestion
                has_color_attachment_count = true;
                break;
```
or print out `color_attachment_count` in error message

Reply:
ya, I just moved this, but agree, let me improve this while here